cmd/subctl/execute/execute.go

```diff
@@ -0,0 +1,64 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package execute
+
+import (
+	"fmt"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	"github.com/submariner-io/submariner-operator/pkg/reporter"
+	"os"
+
+	"github.com/submariner-io/submariner-operator/internal/exit"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/client"
+	"github.com/submariner-io/submariner-operator/pkg/cluster"
+)
+
+func OnMultiCluster(restConfigProducer restconfig.Producer, run func(*cluster.Info, reporter.Interface) bool) {
+	success := true
+
+	status := cli.NewReporter()
+	for _, config := range restConfigProducer.MustGetForClusters() {
+		status.Start("Cluster %q", config.ClusterName)
+
+		clientProducer, err := client.NewProducerFromRestConfig(config.Config)
+		if err != nil {
+			exit.OnErrorWithMessage(err, "Error creating the client producer")
+		}
```

Comment:
We shouldn't exit here if one cluster fails.

```suggestion
                if err != nil {
			status.Error(err, "Error creating the client producer")
			continue
		}
```

Reply:
`status.Error` returns `error` which needs to be checked which is reduntant. Hence `status.Failure` is appropriate here.

Reply:
so unless, error is being returned or checked, `status.Error` can't be used.

Reply:
yeah `status.Failure(fmt.Sprintf("Error creating the client producer: %v", err))` would be OK - that's essentially what `status.Error` does.  The other alternatives are:

`_ = status.Error(err, "Error creating the client producer")`

or

```
err = status.Error(err, "Error creating the client producer")`
if err != nil {
    continue
}
```

Reply:
Used status.Failure